source/module_esolver/esolver_ks_lcao_elec.cpp

```diff
@@ -130,6 +129,37 @@ namespace ModuleESolver
                 this->psi = new psi::Psi<std::complex<double>>(GlobalC::kv.nks, ncol, this->LOWF.ParaV->nrow, nullptr);
             }
         }
+
+        // init Hamiltonian
+        if (this->phami != nullptr)
+        {
+            delete this->phami;
+            this->phami = nullptr;
+        }
+        if(this->phami == nullptr)
+        {
+            // two cases for hamilt class
+            // Gamma_only case
+            if (GlobalV::GAMMA_ONLY_LOCAL)
```

Comment:
I think it would be better to unify the two cases as a single interface of hamilt class.
Also the name 'phami' is not very self-evident, besides it will be easily interpreted as pha / mi, which is extremely confusing for English speakers.
Why not use something like p_hamilt?